runtime/queries/spade/highlights.scm

```diff
@@ -0,0 +1,121 @@
+(self) @variable.builtin
+
+(unit_definition (identifier) @function)
+
+(parameter (identifier) @variable.parameter)
+
+((pipeline_reg_marker) @keyword)
+
+(scoped_identifier
+  path: (identifier) @namespace)
+(scoped_identifier
+ (scoped_identifier
+  name: (identifier) @namespace))
+
+((builtin_type) @type.builtin)
+
+((identifier) @type.builtin
+ (#any-of?
+    @type.builtin
+    "uint"
+    "Option"
+    "Memory"))
+
+((identifier) @constant.builtin
+ (#any-of? @constant.builtin "Some" "None"))
+
+((pipeline_stage_name) @label)
+
+((stage_reference
+    stage: (identifier) @label))
+
+[
+    "pipeline"
+    "let"
+    "set"
+    "entity"
+    "fn"
+    "reg"
+    "reset"
+    "inst"
+    "assert"
+    "struct"
+    "enum"
+    "stage"
+    "impl"
+    "port"
+    "decl"
+    "mod"
+    "where"
+] @keyword
+
+[
+ "use"
+] @keyword.import
+
+[
+    "$if"
+    "$else"
+    "$config"
+] @keyword.directive
+
+((comptime_if  ["{" "}"] @keyword.directive))
+((comptime_else  ["{" "}"] @keyword.directive))
+
+((attribute) ["#" "[" "]"] @punctuation.delimiter)
+
+[
+  "else"
+  "if"
+  "match"
+] @keyword.control.conditional
+
+(bool_literal) @constant.builtin.boolean
+(int_literal) @constant.numeric.integer
+
+[
+  "&"
+  "mut"
+  "-"
+  "=>"
+  ">"
+  "<"
+  "="
+  "->"
+  "~"
+  "!"
+] @operator
+
+
+((op_add) @operator)
+((op_sub) @operator)
+((op_mul) @operator)
+((op_equals) @operator)
+((op_lt) @operator)
+((op_gt) @operator)
+((op_le) @operator)
+((op_ge) @operator)
+((op_lshift) @operator)
+((op_rshift) @operator)
+((op_bitwise_and) @operator)
+((op_bitwise_xor) @operator)
+((op_bitwise_or) @operator)
+((op_logical_and) @operator)
+((op_logical_or) @operator)
+
+
+[
+  (line_comment)
+] @comment @spell
+
+
+((identifier) @type
+  (#match? @type "[A-Z]"))
+
+((scoped_identifier
+    name: (identifier) @type)
+ (#match? @type "^[A-Z]"))
+
+((identifier) @constant
+ (#match? @constant "^[A-Z][A-Z%d_]*$"))
```

Comment:
```suggestion
 (#match? @constant "^[A-Z][A-Z\\d_]*$"))
```

I believe the `%` is used in `#lua-match?` from neovim, here it needs a double escape